src/datomish_user_agent_service/core.cljs

```diff
@@ -23,23 +18,30 @@
 (defonce connection-pair-chan
   (a/promise-chan))
 
+(defn on-js-reload []
+  (js/console.log "on-js-reload"))
+
 (def app (server/app connection-pair-chan))
 
-(defn server [port]
-  ;; This is the secret sauce. you want to capture a reference to the app function (don't use it
-  ;; directly) this allows it to be redefined on each reload this allows you to change routes and
-  ;; have them hot loaded as you code.
-  (doto (.createServer http #(app %1 %2))
-    (.listen port)))
+;; We want to refer to app by name; partial captures the value at
+;; definition time.
+(defn- handle [& rest]
+  (apply app rest))
 
 (defn -main []
+  (.on js/process "uncaughtException" (fn [err]
+                                        (println (aget err "stack"))
+                                        (js/process.exit 101)))
+
+  (.on js/process "unhandledRejection" (fn [reason p]
+                                         (println (aget reason "stack"))
+                                         (js/process.exit 102)))
+
   (a/take!
-    (go-pair
-      (js/console.log "Opening Datomish knowledge-base.")
-      (let [c (<? (d/<connect "")) ;; In-memory for now.
-            _ (<? (d/<transact! c api/tofino-schema))]
-        c))
+    (server/<connect "") ;; In-memory for now.
```

Comment:
Core is used by `lein figwheel dev`; it's a little different than the regular setup to allow hotloading.  We could make this on disk somewhere, but it's quite nice to be able to kill the process to get a new KB, and otherwise just reload to keep your data.